### ISSUE TITLE
Current location of AllowUnsafeBlocks (clr "unsafe") does not work in .NET Core

### ISSUE THREAD

**Issue by ClxS** [premake/premake-core#1549]
The writing of AllowUnsafeBlocks here:
https://github.com/premake/premake-core/blob/b47a949c75d7ad4447558c5204777ff766ad59b7/modules/vstudio/vs2005_dotnetbase.lua#L256

Will need moving to the projectProperties block here:
https://github.com/premake/premake-core/blob/63a1e5ff3b66ef97392f42a8df7da91fc863ecf4/modules/vstudio/vs2005_csproj.lua#L54

This is because the .NET Core format of projects only enables use of unsafe code if it's in the primary PropertyGroup. Entries in conditional ones are ignored.

**Comment by ddobrev** [comment]
This one is extremely important to us as well. https://github.com/premake/premake-core/pull/1551 solves one of our problems and I'm grateful. However, without this one it's in vain to our case.

**Comment by ClxS** [comment]
@ddobrev I probably won't have time to fix it myself in the repo yet, but here's my local override which fixes it.
```lua
local function tryWriteUnsafeBlocks(cfg)
	if cfg.clr == "Unsafe" then
		p.w('<AllowUnsafeBlocks>true</AllowUnsafeBlocks>')
	end
end

premake.override(premake.vstudio.cs2005.elements, "projectProperties", function(base, cfg) 
    local calls = base(cfg);
    table.insert(calls, tryWriteUnsafeBlocks)
    return calls;
end)
```

This wont remove the spurious AllowUnsafeBlocks in the conditional elements, but those are ignored anyway so it's only a problem in that it's a little bit untidy.

**Comment by ddobrev** [comment]
@ClxS thank you, we'll patch our copy. I do look forward to having this fixed upstream, though: manual patching is a burden.

**Comment by starkos** [comment]
Would be happy to approve a pull request.

**Comment by ddobrev** [comment]
@starkos I guess he would first like to clean these useless unsafe blocks up.

**Comment by nickclark2016** [comment]
@starkos do we want this as part of my PR?

**Comment by ddobrev** [comment]
@nickclark2016 I do. 😄

**Comment by nickclark2016** [comment]
Personally, I'd like to keep scope on PRs as narrow as possible. Makes them easier to review in a timely manner.  That said, I'm more than happy to do a 2-for-1 if the main development team wants that.

**Comment by samsinsane** [comment]
Our PR template has a checkbox that states "Focus on a single fix or feature; remove any unrelated formatting or code changes", this isn't an arbitrary requirement. We've had experience with the "2-for-1" PRs, and they're not worth the headaches. I don't know about you guys, but I'm not a huge fan of watching cool features stall because someone felt like poorly fixing an unrelated bug. So, in general, no we don't want you to just add more fixes to a PR.

**Comment by nickclark2016** [comment]
Sounds good. Once #1551 is merged into master, I'll start looking at this one. Probably some time this coming weekend.

**Comment by ddobrev** [comment]
@nickclark2016 since https://github.com/premake/premake-core/pull/1551 has been merged for a week, would you please have an idea when this one will follow?

**Comment by nickclark2016** [comment]
I've been extremely busy lately, but hopefully within the next 2 weeks or so.

**Comment by ddobrev** [comment]
I understand, thank you for your reply.
